### PR TITLE
raidboss/oopsy: add AMR

### PR DESCRIPTION
**Pull request by quisquous**
This is a work in progress of my own personal Another Mount Rokkon triggers. Largely untested.

I am trying to do this content blind for once (scheduling is also hard, so it might be a little) so please do not correct me or tell me if I'm right or wrong or whatever. I will probably not merge these until I'm done, mostly so that I don't have people trying to give me feedback and spoiling me in ways I don't want to be spoiled.

Feel free to grab any part of this and submit your own PR (that I will not look at) if you think it's a good idea.

### CODE REVIEW

**Comment by valarnin** [comment]
I know you've stated you're progging this blind, but it appears you have already progged past the first boss. Do you want the rest of the timeline for it?

**Comment by quisquous** [comment]
> I know you've stated you're progging this blind, but it appears you have already progged past the first boss. Do you want the rest of the timeline for it?

Thanks for asking, but at the moment no!

**Comment by Akurosia** [comment]
on first boss, during the blue shadow phase (find safespot + stack/spread) it sometimes calls the wrong spot for me. Most times when the first safespot is north, it calls south for me (not sure if this is 100% of times but most of them i would say)

**Comment by Akurosia** [comment]
I had yesterday again the case where safespot north got called out as south (and as i saw the boss3 timelines i am sure that i compiled the latest version after the fix)
image

Just for reference these are the replacements i made in ACT itself for this trigger: 
image

File: ui/raidboss/data/06-ew/dungeon/another_mount_rokkon.ts [hunk 1]
```diff
@@ -0,0 +1,1100 @@
...
+        if (matches.target === data.me)
+          return { alarmText: output.chargeOnYou!() };
+        return { alertText: output.chargeOn!({ player: data.ShortName(matches.target) }) };
+      },
+    },
+    {
+      id: 'AMR Shishu Raiko Howl',
+      type: 'StartsUsing',
+      netRegex: { id: '8654', source: 'Shishu Raiko', capture: false },
+      response: Responses.bleedAoe('info'),
```
Comment (Akurosia) [inline ui/raidboss/data/06-ew/dungeon/another_mount_rokkon.ts:409]: From my expirience, this one is just an AoE.

**Comment by quisquous** [comment]
> I am trying to do this content blind for once (scheduling is also hard, so it might be a little) so please do not correct me or tell me if I'm right or wrong or whatever. I will probably not merge these until I'm done, mostly so that I don't have people trying to give me feedback and spoiling me in ways I don't want to be spoiled.

Sorry, I know this is awkward but I'm trying to do this blind (and with bad scheduling, we've only gone twice ;;) because I don't get to do that many things blind.  I was trying to put this draft PR up because I was trying to be slightly helpful and didn't want folks to bother duplicating work that I was doing for my own use if they want to grab things that I had done. However, I really don't want feedback on this about things that are wrong. Feel free to write down anything you think is wrong or not working and I'll take a look once we're done and I'm ready to land this.

**Comment by quisquous** [comment]
Sorry for the super delay, but I think this is ready to go.

I took a look through PF strats vaguely and I think the biggest difference is second boss second towers, where we had different strats for if the debuffs were "full mixed" (everybody has two colors) and "half mixed" (some people have two colors some have all the same colors), but it seems like other people maybe have different strats and so I'm not sure what should be called out here.

@xiashtra @Akurosia @valarnin @g0shu Did y'all have any additional feedback about this before I merge it?

**Comment by Akurosia** [comment]
Maybe call out if at least 1 person has 3 of the same colored buffs, as from my expirience that is what most groups do (have 2 different strats depending if all player have blue and red buffs or if 3 same colored debuffs are available)

Other than that most of the stuff was working for me when i tried it a few days ago and i did not noticed any major flaw in callouts.

**Comment by xiashtra** [comment]
> Sorry for the super delay, but I think this is ready to go.
> 
> I took a look through PF strats vaguely and I think the biggest difference is second boss second towers, where we had different strats for if the debuffs were "full mixed" (everybody has two colors) and "half mixed" (some people have two colors some have all the same colors), but it seems like other people maybe have different strats and so I'm not sure what should be called out here.
> 
> @xiashtra @Akurosia @valarnin @g0shu Did y'all have any additional feedback about this before I merge it?

The PF strat I've seen for Malformed Reincarnation (the final towers) uses the same strat regardless of number of same colored buffs.  This is likely a situation where there's completely different strats floating around.

I'm not sure if there's any remaining issues still present.  I might get a chance to look this weekend.

**Comment by quisquous** [comment]
> Maybe call out if at least 1 person has 3 of the same colored buffs, as from my expirience that is what most groups do (have 2 different strats depending if all player have blue and red buffs or if 3 same colored debuffs are available)
> 
> Other than that most of the stuff was working for me when i tried it a few days ago and i did not noticed any major flaw in callouts.

FWIW, our strat was something like this: https://raidplan.io/plan/9tVR4vj9kPjgF3PM

* if "half-mixed" (i.e. half the group has mixed colors, i.e. some people have all red or blue):
  * look at your first tower color, go to the quadrant with two towers of that color
  * no flexing, doesn't matter who is where on that side
  * "easy mode"
* if "full mixed" (i.e. everybody has mixed colors):
  * go to quadrants with one tower of each color instead
  * need to flex to make sure you have one of each color in each mixed quadrant
  * also need to stand opposite the first tower you have
 
I think there's no way around the place tower -> cross over to inside tower 1 -> outer corner tower 2-> placed tower 3 dance, it's just about where you start.

But I saw https://www.youtube.com/watch?v=TzoNEWbMpQ0#t=7m53s which seems like a different setup that always flexes and gets people on mixed quadrants always

**Comment by xiashtra** [comment]
> > Neither one of these is what I've seen in pf 😬.
> 
> Do you have a link to a PF strat?

I'll see if I can find something, but honestly, it's just skipped most of the time.

**Comment by xiashtra** [comment]
> > Neither one of these is what I've seen in pf 😬.
> 
> Do you have a link to a PF strat?

This is what I've seen: https://ff14.toolboxgaming.space/?id=938195953989861&preview=1  No adjusts regardless of full- or half-mixed.

**Comment by quisquous** [comment]
Ok, so sounds like a basic strategy that could account for most of this would say something like:
* alertText: drop blue, first orange
* infoText: (all mixed colors) / (half mixed colors)

or something like this

**Comment by xiashtra** [comment]
> EDIT: I don't think any strat cares about the color you are dropping, so maybe:
> 
> * alertText: blue -> blue -> orange (half mixed)

I think that would probably satisfy everyone.

**Comment by ViperFE** [comment]
Is there a way to setup the final boss shadow clone mechanic using this simplier strat? 
https://www.youtube.com/watch?v=KghjI7xn6X0
It keeps calling out my position wrong regardless of what Role > Stack thing I set.

**Comment by quisquous** [comment]
Could you file a separate issue for this rather than a comment on a pull request that I might lose track of?

> Is there a way to setup the final boss shadow clone mechanic using this simpler Strat?

Maybe there's some confusion. What do you think the cactbot strat is doing vs this strat? (Which mechanic specifically do you mean here? I thought you meant soldiers of death, but then you talk about "final arrow mechanic" after, which makes me think that's not true.

> Same for the final arrow mechanic, keeps telling me to go to weird places. We use role-based positions for the last one. Tanks South or North, DPS East or West.

Sorry for the confusion. It doesn't tell you where to go. It tells you what tether you have and who should be together.

**Comment by ViperFE** [comment]
Sorry I'm new to github so I don't know exactly where to post things. I also don't know really how to explain it other than how I already have. I'll try though. 

In the video the first part I was talking about starts at 2:55. I'm either S or W and if there's any adjustment, the healer and other DPS swaps. Oftentimes it tells me to pair with the wrong person or I don't even know what info it's trying to give because I don't know the strat that it's trying to use at all. The bait callout for tether doesn't work either, I think it's assuming I'm somewhere else.

For the second part 5:07, soldiers of death. It's telling me to go where the tanks are supposed to go. We do role based regardless of tethers, So tanks would take south and north safe spots, DPS on east and west. I'm always with the other DPS so it shouldn't even be telling me any info regarding who to be with, just where to go.

**Comment by quisquous** [comment]
In general, cactbot isn't assuming any strat at all, and is just telling you what you have / what other people have, if that helps.

> In the video the first part I was talking about starts at 2:55. I'm either S or E and if there's any adjustment, the healer and other DPS swaps. Oftentimes it tells me to pair with the wrong person or I don't even know what info it's trying to give because I don't know the strat that it's trying to use at all. The bait callout for tether doesn't work either, I think it's assuming I'm somewhere else.

Yeah, I wonder if the callout is just confusing. In this case, it tells you who has the same thing you do (in case you are flexing). So if you and somebody else have tether, it'll say "Back Tether (w/Somebody Else)" or something. This one (for whatever reason) never got switched over to use the melee/roles logic. It's similar to triggers  earlier where it will tell you that you have blue tower or spread w/ another player. That person is not the person you stack with, but the person who has the same thing you do.

What isn't working about the bait tether?

> For the second part 5:07, soldiers of death. It's telling me to go where the tanks are supposed to go.

This call doesn't tell you to go anywhere, it's trying to tell you what you have. It will say "Left (roles together)" meaning you have the left safe tether and you should be on the cardinal with your role partner. You can use this info with any pre-positioning.

**Comment by ViperFE** [comment]
Ahh I see I had assumed "AMR Moko Soldiers of Death Blue Add" was a safespot callout, since he has 8 spawns and none of them are NESW cardinals to match the call. His arrow distance is also dictated by which side he spawns in, N/S always shoots short, E/W always shoots long. So you can often find the safespots as soon as he spawns, I imagine cactbot could figure this out for you easier and be able to call safespots?

Tethers on the last mechanic definitely aren't random, or in the very least the angles it gives you aren't.  you can always 100% pair with your fellow DPS/Support, find which of your two safespots is or isn't safe and go to it. You'll always be in a pair that can show your safe angle towards mid after facing away from each other. This is why I was confused that cactbot wasn't telling me which direction my second cleave would face, since that's really the only info you might miss after you've found your safespot.

For the bait tether I am just reading it wrong. I just noticed it's telling you both mechanics on your side in reverse order depending on what you get, but I might just edit mine to only say where tether needs to be and obviously I'm the opposite of it if I'm not tether. We'll be doing a few clears tonight and I'll keep an eye on it.

**Comment by xiashtra** [comment]
> I am nearly positive this is not true. We were doing roles together by default and there were a couple of times where we had to flex because the two dps both had lefts and the tank and healer both had rights. In true ffxiv rng, mooostly it was roles but there were a couple where it wasn't.

Soldiers of Death tethers should always be support pair and dps pair, no flex needed.  I suppose it might scramble if there's a badly timed death, like some other mechanics do, but for a clean run, role pairs should always work.

**Comment by quisquous** [comment]
> Soldiers of Death tethers should always be support pair and dps pair, no flex needed. I suppose it might scramble if there's a badly timed death, like some other mechanics do, but for a clean run, role pairs should always work.

Oh, I see what the confusion is on my part. Our healer and dps swapped roles midway through, but we continued to call it airquotes "roles" stacks for our original jobs and I forgot about the airquotes part. The blue add having a set long and short distance checks out from video, so I'll add that in.

**Pull request merged**